x-pack/plugins/security_solution/public/timelines/components/timeline/search_or_filter/index.tsx

```diff
@@ -110,19 +112,48 @@ const StatefulSearchOrFilterComponent = React.memo<Props>(
 
     const arrDataView = useMemo(() => (dataView != null ? [dataView] : []), [dataView]);
 
+    // Keep filter manager in sync with redux filters
+    useEffect(() => {
+      if (!deepEqual(filterManager.getFilters(), filters)) {
+        filterManager.setFilters(filters);
+      }
+    }, [filterManager, filters]);
+
+    // When a filter update comes in through the filter manager, update redux
+    useEffect(() => {
+      let isSubscribed = true;
+      const subscriptions = new Subscription();
+
+      subscriptions.add(
+        filterManager.getUpdates$().subscribe({
+          next: () => {
+            if (isSubscribed) {
+              const filtersWithoutDropArea = getNonDropAreaFilters(filterManager.getFilters());
+              if (!deepEqual(filtersWithoutDropArea, filters)) {
+                setFilters({
+                  id: timelineId,
+                  filters: filtersWithoutDropArea,
+                });
+              }
+            }
+          },
+        })
+      );
+
+      return () => {
+        isSubscribed = false;
+        subscriptions.unsubscribe();
+      };
```

Comment:
this code seems extremely complex. I'm not sure I understand what we're trying to do here with the subscription, the `isSubscribe` set to `true` then `false` and we unsubscribe at the same time?

Can you explain what this code is doing? There has to be an easier way to do this, just with the `getUpdates$` observable...

Reply:
I cut this code here from another component further down the tree and didn't touch it. But yeah, it's quite cumbersome. `isSubscribed` is a safeguard, probably stemming from years ago when these updates on potentially unmounted components can wreck your entire app. The assumption here must have been that the callback could be called on an unmounted component __before__ the subscription had been unregistered.

There was another comment about this here: https://github.com/elastic/kibana/pull/180850#discussion_r1567444178 and it's been since simplified.

Reply:
that makes sense, thanks for the explanation! The simplified code is easier to ready, thank you! :)